packages/veritone-widgets/src/widgets/ContentTemplate/index.js

```diff
@@ -0,0 +1,108 @@
+import React from 'react';
+import { get } from 'lodash';
+import { objectOf, any, func } from 'prop-types';
+import { ContentTemplate } from 'veritone-react-common';
+
+import widget from '../../shared/widget';
+
+class ContentTemplateWidget extends React.Component {
+  static propTypes = {
+    templateData: objectOf(any).isRequired,
+    initialTemplates: objectOf(any),
+    handleUpdateContentTemplates: func.isRequired
+  };
+
+  static defaultProps = {
+    initialTemplates: {}
+  };
+
+  state = {
+    contentTemplates: {}
+  };
+
+  // eslint-disable-next-line react/sort-comp
+  UNSAFE_componentWillMount() {
+    const newState = {
+      contentTemplates: { ...this.props.initialTemplates }
+    };
+
+    this.setState(newState);
+  }
+
+  onListChange = (templateSchemaId, remove = false) => {
```

Comment:
what's the reason for not separating this out into `onAdd & onRemove`? ideally you should be writing functions with a single purpose